### ISSUE TITLE
How to use this model?

### ISSUE THREAD

**Issue by XXXTJXX** [USDA-ARS-ACSL/ExcelInterface_Gossym#1]
I just have no idea how to use this model, and I would be grateful for any help you can give me on using steps.

**Comment by DennisTimlin-USDA** [comment]
Thanks for your interest.  This is just the code for the original GOSSYM for anyone who was interested. It is best to use the newer version - GOSSYM-2DSOIL. There is also an interface for it - I think the repository is ExelInterface for Gossym

Let me know what you are interested in using the model for and I can help you get it working. I have documentation for the excel interface in the Maizesim repository, I will copy it to the gossym one. I will also find the gossym documentation.

Dennis

**Comment by XXXTJXX** [comment]
I'm currently trying to use the GOSSYM model and some field trial data to model and simulate the morphology and growth process of cotton above ground. The preliminary plan is to divide the aboveground part of cotton into four parts: internode, leaf, petiole and boll for morphological simulation, then use the water balance module and biomass competition module to complete the simulation of the aboveground part of cotton affected by water stress, and finally display it visually.

The first and foremost problem is that I don't know much about the basic structure, function and usage of GOSSYM model at the moment, so I would like to seek your help, and at the same time, I will be grateful if you can make some suggestions for my project.

**Comment by XXXTJXX** [comment]
And thanks for your reply!

**Comment by DennisTimlin-USDA** [comment]
I had to strip out the documentation I added as attachments, I will put them on the github site

Dennis

From: Timlin, Dennis - REE-ARS
Sent: Tuesday, March 12, 2024 12:04 PM
To: USDA-ARS-ACSL/ExcelInterface_Gossym ***@***.***>
Cc: Beegum, Sahila - REE-ARS ***@***.***>
Subject: RE: [USDA-ARS-ACSL/ExcelInterface_Gossym] How to use this model? (Issue #1)

Sounds interesting, we will be glad to help. Are you with a University?

I am copying Dr. Sahila Beegum who is a post-doc working with the model now.


I have a better approach for you. We have developed a GUI (CLASSIM) for our crop models that removes all the tedious file accumulation needed to run the models. I put the source code for the model on github for people who are interested in getting into the code. There are two interfaces, CLASSIM which is good for doing a limited number of simulations (due to the work in creating the simulations using a dialog dropdowns and text. If you want to do 30 -100 or more year analysis or large spatial scale simulations then the excel interface is better because you can fill in the simulation details using scripts. We have done 20, 30 50 thousand sims using the excel interface.

CLASSIM is great to start with as it has a lot of graphics for learning the model and you can get weather and soil data from online databases. Classim is on Github but the version is about a month behind. The newest version is on box:


The setup program will install everything - unfortunately it is windows based. At some point we will port to linux - it is written in python.

You can get it here - it is an installer.
CLASSIM and Documentation | Powered by Box<https://ars-usda.app.box.com/s/vtf2ghvi736slhnvtc2u4t6xan04ci9w>

***@***.***

There may be some problems so keep in touch and we will work them out. Sahila and I would be glad to help flesh out your project as well.

Best wishes,
Dennis

**Comment by XXXTJXX** [comment]
Yes, I am currently a college student and I am working for my bachelor's thesis. I'm glad to get your help.

Your advice is very useful to me. I will try my best. At the same time, thank you for sharing. After the program is installed and explored in person, I will share my thoughts and questions with you.

Finally, I would like to thank you and Sahila again for your help. I am very willing and hope to keep in touch with you.

**Comment by DennisTimlin-USDA** [comment]
If you use onedrive to backup your windows documents folder we may have to make some manual adjustments. The interface has instructional video links for some of the tabs - we are adding more. There is also documentation but it needs a bit of updating for the screen shots

Try the beou maize simulation first to make sure everything is working.

Dennis

**Comment by XXXTJXX** [comment]
Sure, I will try it. And looking forward to our follow-up contact.



Best wishes,
TJ




------------------ Original ------------------
From: Dennis Timlin ***@***.***&gt;
Date: Wed,Mar 13,2024 1:00 AM
To: USDA-ARS-ACSL/ExcelInterface_Gossym ***@***.***&gt;
Cc: XXXTJXX ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [USDA-ARS-ACSL/ExcelInterface_Gossym] How to use this model?(Issue #1)





 If you use onedrive to backup your windows documents folder we may have to make some manual adjustments. The interface has instructional video links for some of the tabs - we are adding more. There is also documentation but it needs a bit of updating for the screen shots
 
 Try the beou maize simulation first to make sure everything is working.
 
 Dennis
